src/structlog/stdlib.py

```diff
@@ -365,3 +365,74 @@ def render_to_log_kwargs(wrapped_logger, method_name, event_dict):
         "msg": event_dict.pop("event"),
         "extra": event_dict,
     }
+
+
+class ProcessorFormatter(logging.Formatter):
+    """
+    Call ``structlog`` processors on :mod:`logging.LogRecord`\ s.
+
+    This :class:`logging.Formatter` allows to configure :mod:`logging` to call
+    *processor* on ``structlog``-borne log entries (origin is determined solely
+    on the fact whether the ``msg`` field on the :class:`logging.LogRecord` is
+    a dict or not).
+
+    This allows for two interesting use cases:
+
+    #. You can format non-``structlog`` log entries.
+    #. You can multiplex log records into multiple :class:`logging.Handler`\ s.
+
+    Please refer to :doc:`standard-library` for examples.
+
+    :param callable processor: A ``structlog`` processor.
+    :param foreign_pre_chain:
+        If not `None`, it is used as an iterable of processors that is applied
+        to non-``structlog`` log entries before *processor*.  If `None`,
+        formatting is left to :mod:`logging`. (default: `None`)
+
+    :rtype: str
+
+    .. versionadded:: 17.1.0
+    """
+    def __init__(self, processor, foreign_pre_chain=None, *args, **kwargs):
+        fmt = kwargs.pop("fmt", "%(message)s")
+        super(ProcessorFormatter, self).__init__(*args, fmt=fmt, **kwargs)
+        self.processor = processor
+        self.foreign_pre_chain = foreign_pre_chain
+
+    def format(self, record):
+        """
+        Extract ``structlog``'s `event_dict` from ``record.msg`` and format it.
+        """
+        if isinstance(record.msg, dict):
+            logger = record._logger
+            meth_name = record._name
+            # We need to copy because it's possible that the same record gets
+            # processed by multiple logging formatters.
+            ed = record.msg.copy()
+        else:
+            if self.foreign_pre_chain is None:
+                # Let logging format the message.
+                return super(ProcessorFormatter, self).format(record)
+            # Non-structlog allows to run through a chain to prepare it for the
+            # final processor (e.g. adding timestamps and log levels).
+            ed = {"event": record.getMessage()}
+            logger = None
+            meth_name = record.levelname.lower()
+            for proc in self.foreign_pre_chain:
+                ed = proc(None, meth_name, ed)
+
+        return self.processor(logger, meth_name, ed)
```

Comment:
You are not actually formatting the message anywhere in this method.
So if you have a logging configuration like `'format': u'\n%(message)s [in %(pathname)s:%(lineno)d]',` you are not respecting that. And the end format is always the plain message only.